platforms/core-runtime/classloaders/src/main/java/org/gradle/internal/classloader/VisitableURLClassLoader.java

```diff
@@ -183,21 +181,23 @@ public void transformFailed(@Nullable String className, Throwable cause) {
 
         @Override
         protected Class<?> findClass(String name) throws ClassNotFoundException {
-            errorHandler.enterClassLoadingScope(name);
-            Class<?> loadedClass;
             try {
-                loadedClass = super.findClass(name);
+                errorHandler.enterClassLoadingScope(name);
+                return super.findClass(name);
             } catch (Throwable e) {
                 throw errorHandler.exitClassLoadingScopeWithException(e);
+            } finally {
+                errorHandler.exitClassLoadingScope();
```

Comment:
"first" things "first".  this should ensure its really gonna be called.

Reply:
> "first" things "first". this should ensure its gonna be called, no matter what.

:x: This doesn't look correct. The code should call either `exitClassLoadingScopeWithException` or `exitClassLoadingScope`. Your change makes it call both, which is likely benign, as the pending exception has been cleared by `exitClassLoadingScopeWithException`, but also useless and may trigger subtle bugs if the behavior of the errorHandler evolves.